test/tree-sitter/ExpressionBuilder.test.ts

```diff
@@ -110,4 +111,89 @@ defineParameterType({
       ['a {uuid}', 'a {date}', /a regexp/]
     )
   })
+
+  it('builds expressions from C# source', async () => {
+    const stepdefs: Source = {
+      language: 'c_sharp',
+      content: `
+namespace Bowling.SpecFlow.StepDefinitions
+{
+    [Binding]
+    public class BowlingSteps
+    {
+        private readonly BowlingDriver _bowlingDriver;
+
+        public BowlingSteps(BowlingDriver bowlingDriver)
+        {
+            _bowlingDriver = bowlingDriver;
+        }
+
+        [Given(@"a new bowling game")]
+        public void GivenANewBowlingGame()
+        {
+            _bowlingDriver.NewGame();
+        }
+
+        [When(@"all of my balls are landing in the gutter")]
+        public void WhenAllOfMyBallsAreLandingInTheGutter()
+        {
+            _bowlingDriver.Roll(0, 20);
+        }
+
+        [Then(@"my total score should be (\\d+)")]
+        public void ThenMyTotalScoreShouldBe(int score)
+        {
+            _bowlingDriver.CheckScore(score);
+        }
+
+        [StepDefinition(@"I roll (\\d+) and (\\d+)")]
+        public void WhenIRoll(int pins1, int pins2)
+        {
+            _bowlingDriver.Roll(pins1, pins2, 1);
+        }
+
+        [When(@"I roll the following series:(.*)")]
+        public void WhenIRollTheFollowingSeries(string series)
+        {
+            _bowlingDriver.RollSeries(series);
+        }
+
+        [When(@"I roll")]
+        public void WhenIRoll(Table rolls)
+        {
+            _bowlingDriver.RollSeries(rolls);
+        }
+    }
+}
+`,
+    }
+
+    const parameterTypes: Source = {
+      language: 'c_sharp',
+      content: `
+[Binding]
+public class Transforms
+{
+    [StepArgumentTransformation(@"in (\\d+) days?")]
+    public DateTime InXDaysTransform(int days)
+    {
+      return DateTime.Today.AddDays(days);
+    }
+
+    [StepArgumentTransformation]
+    public XmlDocument XmlTransform(string xml)
+    {
+       XmlDocument result = new XmlDocument();
+       result.LoadXml(xml);
+       return result;
+    }
+}`,
+    }
+
+    const expressions = expressionBuilder.build([stepdefs, parameterTypes], [])
+    assert.deepStrictEqual(
+      expressions.map((e) => e.source),
+      ['I have {int} cukes in my belly', 'you have some time', 'a {iso-date}', 'a {date}']
```

Comment:
The C# code above doesn't contain any step definitions with these Cucumber Expressions. The C# code must be changed in order to make this test pass.

Reply:
Indeed the tests still need work. I essentially stopped coding when I was unable to build the WASM files and left a few things in more of a Draft state. I'll merge main and make the changes.

Reply:
I've updated the tests now